### PR TITLE
Prevent 'truffle create' from overwriting existing files

### PR DESCRIPTION
**Pull request by dmihal**
Discussed in trufflesuite/truffle#809

`truffle create` will no longer overwrite existing files, unless passed the -f flag

### CODE REVIEW

File: lib/create.js [hunk 1]
```diff
@@ -53,23 +53,39 @@ var toUnderscoreFromCamel = function(string) {
...
+  contract: function(directory, name, options, callback) {
+    if(typeof options == "function") {
+      callback = options;
+    }
+
     var from = templates.contract.filename;
     var to = path.join(directory, name + ".sol");
 
+    if (!options.force && fs.existsSync(to)) {
+      console.error('Can not create ' + name + '.sol: file exists');
```
Comment (gnidan) [inline lib/create.js]: Just a note: normal convention is to `callback(err)` and let "upstream processing" handle the outputting. At the very least, `config.logger` should be used in place of `console`. But I think in this case `callback(err)` makes more sense.

Comment (dmihal) [reply lib/create.js]: Updated!

File: lib/create.js [hunk 2]
```diff
@@ -53,23 +53,39 @@ var toUnderscoreFromCamel = function(string) {
...
+  contract: function(directory, name, options, callback) {
+    if(typeof options == "function") {
+      callback = options;
+    }
+
     var from = templates.contract.filename;
     var to = path.join(directory, name + ".sol");
 
+    if (!options.force && fs.existsSync(to)) {
+      return callback('Can not create ' + name + '.sol: file exists');
```
Comment (gnidan) [inline lib/create.js]: Ah shoot, sorry to ask for another set of changes - I just looked and other invocations of `callback(err)` actually pass an error object; mind changing these to `callback(new Error('Can not create...'))`?

Comment (dmihal) [reply lib/create.js]: Good catch! I've added the Error object.

**Pull request merged**